File: python/compound-interest.py
def calculate_compound_interest(principal, time, rate, n):
    amount = principal * (1 + rate / (100 * n)) ** (n * time)
    compound_interest = amount - principal
    print(f"Compound Interest: {compound_interest:.2f}")

# Taking inputs
principal = float(input("Enter Principal Amount: "))
time = float(input("Enter Time (in years): "))
rate = float(input("Enter Rate of Interest: "))
n = int(input("Enter Number of Times Interest Compounded per Year: "))

# Calculation
calculate_compound_interest(principal, time, rate, n)

File: C/compound_interest.c
#include <stdio.h>
#include <math.h>

void calculate_compound_interest(float principal, float time, float rate, int n) {
    float amount = principal * pow((1 + rate / (100 * n)), n * time);
    float compound_interest = amount - principal;
    printf("Compound Interest: %.2f\n", compound_interest);
}

int main() {
    float principal, time, rate;
    int n;

    // Display heading
    printf("========================================\n");
    printf("       COMPOUND INTEREST CALCULATOR     \n");
    printf("========================================\n");

    printf("Enter Principal Amount: ");
    scanf("%f", &principal);
    printf("Enter Time (in years): ");
    scanf("%f", &time);
    printf("Enter Rate of Interest: ");
    scanf("%f", &rate);
    printf("Enter Number of Times Interest Compounded per Year: ");
    scanf("%d", &n);

    // Calculate and display result
    calculate_compound_interest(principal, time, rate, n);

    return 0;
}
